README.md

```diff
@@ -1 +1,72 @@
-# usePolling
+# use-raf-polling
+
+## Overview
+
+*use-raf-polling* is a lightweight npm package that provides a simple and intuitive hook for API polling in React applications. The package simplifies the process of retrieving the fresh data in your applications, making them more responsive and engaging for your users.
+
+## Features and benefits
+
+- Easy to use: *use-raf-polling* provides a straightforward way to implement polling in your React components, with no complex configuration required.
+- Flexible: you can customize the polling interval, polling condition, callback functions to suit your specific use case.
+- Efficient: *use-raf-polling* uses requestAnimationFrame to ensure smooth and performant polling, without blocking the main thread.
+- Reliable: the hook includes error handling and retry logic.
+
+## Usage
+
+Here's a basic example of how to use *use-raf-polling*:
+
+```typescript
+import { usePolling, UsePollingProps } from 'react-polling-hook'
+
+const MyComponent = () => {
+  const pollingProps: UsePollingProps = {
+    callback: async () => {
+      // Your polling logic here
+    },
+    condition: true, // Use this property to start/stop polling. If set to 'true', the API polling will start immediately 
+    interval: 5000, // Set this to your desired polling interval in milliseconds
+    onPollingFailed: () => {
+      console.error("Polling failed!") // Handle failed polling requests here
+    },
+    onPollingSucceed: () => {
+      console.log("Polling succeeded!") // Handle successful polling requests here
+    }
+  }
+  const {
+    isPolling,
+    restartPolling,
+    startPolling,
+    stopPolling
+  } = usePolling(pollingProps)
+
+  return (
+    <div>
+      <p>Polling is currently {isPolling ? 'enabled' : 'disabled'}</p>
+      <button onClick={startPolling}>Start polling</button>
+      <button onClick={stopPolling}>Stop polling</button>
+      <button onClick={restartPolling}>Restart polling</button>
+    </div>
+  )
+}
+```
+
+For more information of usage, please refer to the [example](https://github.com/example/example).  
+Also, check the real [demo app](https://epam.github.io/deps-fe-usePolling/index.html).
+
+## Installation
+
+To install *use-raf-polling* in your project, simply run:
+
+```bash
+# If you use npm:
+npm install use-raf-polling
+
+# Or if you use Yarn:
+yarn add use-raf-polling
+```
```

Comment:
@Rob-n-Dude could u copy\paste this readme to NPM registry after review?

Reply:
discussed that we need to publish changes

Reply:
Try to update it by yourself - based on @Rob-n-Dude RB documentation

Reply:
yeah, it's required to publish new version. to applyreadme changes ;)

feel free to reach me out.  if you face some issues ^_^